db/migrate/20191007125535_mark_withdrawn_courses_as_withdrawn.rb

```diff
@@ -0,0 +1,19 @@
+class MarkWithdrawnCoursesAsWithdrawn < ActiveRecord::Migration[6.0]
+  def up
+    say_with_time "Marking all courses with site statuses that have been suspended with no vacancies" do
+      courses = Course.where(site_statuses: SiteStatus.where(vac_status: "no_vacancies", status: "suspended"))
+      published_courses = courses.select(&:is_published?)
+      courses_with_all_suspended_no_vacancy_sites = published_courses.select do |course|
+        course.site_statuses.all? do |site_status|
+          site_status.vac_status == "no_vacancies" && site_status.status == "suspended"
```

Comment:
Would prefer that these used the enum-created predicates:

```suggestion
          site_status.no_vacancies? && site_status.status_suspended?
```

Assuming, of course, this wasn't avoided intentionally. But will still give a pass as this is a (hopefully) short-lived migration that won't be run again after it's deployed.